src/status_im/browser/permissions.cljs

```diff
@@ -33,29 +60,38 @@
                                                          :message-id message-id}}))))
 
 (fx/defn permission-show-permission
-  [{:keys [db] :as cofx} dapp-name permission message-id yield-control?]
-  {:db (assoc-in db [:browser/options :show-permission]
-                 {:requested-permission permission
-                  :message-id           message-id
-                  :dapp-name            dapp-name
-                  :yield-control?       yield-control?})})
+  [{:keys [db] :as cofx} dapp-name permission message-id yield-control? params]
+  (let [result (cond-> {:requested-permission permission
+                        :message-id           message-id
+                        :dapp-name            dapp-name
+                        :yield-control?       yield-control?}
+                 (not (nil? params)) (assoc :params params))]
+    {:db (assoc-in db [:browser/options :show-permission] result)}))
 
 (defn get-permission-data [cofx allowed-permission]
   (let [multiaccount (get-in cofx [:db :multiaccount])]
     (get {constants/dapp-permission-contact-code (:public-key multiaccount)
           constants/dapp-permission-web3         [(:dapps-address multiaccount)]}
          allowed-permission)))
 
-(fx/defn send-response-to-bridge
-  "Send response to the bridge. If the permission is allowed, send data associated
+(fx/defn send-response
+  "Send response. If the permission is allowed, send data associated
    with the permission"
-  [{:keys [db] :as cofx} permission message-id allowed? data]
-  {:browser/send-to-bridge (cond-> {:type       constants/api-response
-                                    :isAllowed  allowed?
-                                    :permission permission
-                                    :messageId  message-id}
-                             allowed?
-                             (assoc :data data))})
+  [{:keys [db] :as cofx} permission message-id allowed? data params]
+  (let [result (cond-> {:isAllowed allowed?
+                        :permission permission
+                        :messageId message-id}
+                 allowed?
+                 (assoc :data data)
+                 (not (nil? params))
+                 (assoc :params params))]
+    (cond
+      (switch-ethereum-chain-permission? permission)
```

Comment:
because i want to reuse [permission panel](https://github.com/qfrank/status-react/blob/d4ecbd6183ed67192e56e9953d8efca5b197cb44/src/status_im/ui/screens/browser/permissions/views.cljs#L45)